centreon/www/class/centreonDowntime.class.php

```diff
@@ -519,67 +499,55 @@ public function getForEnabledServicegroups()
+            JOIN service s ON sgr.service_service_id = s.service_id
+            JOIN host h ON sgr.host_host_id = h.host_id
+            JOIN servicegroup sg ON sgr.servicegroup_sg_id = sg.sg_id
+            WHERE sg.sg_activate = '1'
             UNION DISTINCT
                 SELECT dt.dt_id,
-                    dt.dt_activate,
-                    dtp.dtp_start_time,
-                    dtp.dtp_end_time,
-                    dtp.dtp_day_of_week,
-                    dtp.dtp_month_cycle,
-                    dtp.dtp_day_of_month,
-                    dtp.dtp_fixed,
-                    dtp.dtp_duration,
-                    h.host_id,
-                    h.host_name,
-                    s.service_id,
-                    s.service_description,
-                    s.service_register
-                FROM downtime_period dtp,
-                    downtime dt,
-                    downtime_servicegroup_relation dtr,
-                    host_service_relation hsr,
-                    hostgroup_relation hgr,
-                    service s,
-                    host h,
-                    servicegroup_relation sgr
-                WHERE dtp.dt_id = dtr.dt_id
-                    AND dtp.dt_id = dt.dt_id
-                    AND dtr.sg_sg_id = sgr.servicegroup_sg_id
-                    AND sgr.hostgroup_hg_id IS NOT NULL
-                    AND sgr.hostgroup_hg_id = hsr.hostgroup_hg_id
-                    AND hsr.service_service_id = s.service_id
-                    AND hsr.hostgroup_hg_id = hgr.hostgroup_hg_id
-                    AND hgr.host_host_id = h.host_id;
+                       dt.dt_activate,
+                       dtp.dtp_start_time,
+                       dtp.dtp_end_time,
+                       dtp.dtp_day_of_week,
+                       dtp.dtp_month_cycle,
+                       dtp.dtp_day_of_month,
+                       dtp.dtp_fixed,
+                       dtp.dtp_duration,
+                       h.host_id,
+                       h.host_name,
+                       s.service_id,
+                       s.service_description,
+                       s.service_register
+                FROM downtime_period dtp
+                JOIN downtime dt ON dtp.dt_id = dt.dt_id
+                JOIN downtime_servicegroup_relation dtr ON dtp.dt_id = dtr.dt_id
+                JOIN servicegroup_relation sgr ON dtr.sg_sg_id = sgr.servicegroup_sg_id
+                JOIN host_service_relation hsr ON sgr.hostgroup_hg_id = hsr.hostgroup_hg_id
+                JOIN hostgroup_relation hgr ON hsr.hostgroup_hg_id = hgr.hostgroup_hg_id
+                JOIN service s ON hsr.service_service_id = s.service_id
+                JOIN host h ON hgr.host_host_id = h.host_id
+                WHERE sgr.hostgroup_hg_id IS NOT NULL;
         SQL;
 
-        $statement = $this->db->query($request);
+        $statement = $this->db->prepare($request);
+        $statement->execute();
```

Comment:
There's nothing to prepare here.
```suggestion
        $statement = $this->db->query($request);
```